zen3geo/datapipes/pystac.py

```diff
@@ -0,0 +1,92 @@
+"""
+DataPipes for :doc:`pystac <pystac:index>`.
+"""
+from typing import Any, Dict, Iterator, Optional
+
+try:
+    import pystac
+except ImportError:
+    pystac = None
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+
+@functional_datapipe("read_from_pystac_item")
+class PySTACItemReaderIterDataPipe(IterDataPipe):
+    """
+    Takes files from local disk or URLs (as long as they can be read by pystac)
+    and yields :py:class:`pystac.item.Item` objects (functional name:
+    ``read_from_pystac_item``).
```

Comment:
Maybe `read_to_pystac_item` instead?

```suggestion
@functional_datapipe("read_to_pystac_item")
class PySTACItemReaderIterDataPipe(IterDataPipe):
    """
    Takes files from local disk or URLs (as long as they can be read by pystac)
    and yields :py:class:`pystac.item.Item` objects (functional name:
    ``read_to_pystac_item``).
```

Reply:
Done in d66f1f3ff804a189f4377c2a6b3099e5990fc967